dvc/progress.py

```diff
@@ -74,6 +76,8 @@ def __init__(
         self.desc_persist = desc
         if disable is None:
             disable = logger.getEffectiveLevel() > level
+            if not disable and hasattr(kwargs["file"], "isatty"):
```

Comment:
Need a way to override this, say with env var, for performance testing like in #1970.

Reply:
Also, logic seems not corresponding to anything desirable. If we are redirecting then `disable` flag should be ignored not the other way around.

Reply:
performance testing? `tqdm` won't hurt performance so there's no need to disable it via an env var, but if you mean forcefully *enable* all progress via an an env var to help with profiling - well, maybe.

Reply:
I measure when there is no output for long period of time, so I need pbar to be there, and it would be disabled because I ran dvc with redirected output.

Reply:
isatty still needs to take precedence over all disable flags. The core use case - a user redirects outputs to a file, pbars disappear, which won't work if you code it like that.

Reply:
> isatty still needs to take precedence over all disable flags

not true. As you stated/requested in https://github.com/iterative/dvc/pull/2550/files/fdd2f3aaa714fed4d5c33c7508f67440be36b77c#r329573320 you need an env var to be able to override `isatty`.

> The core use case - a user redirects outputs to a file, pbars disappear, which won't work if you code it like that.

yes, it will work as expected.

Reply:
This is how it supposed to work:

1. If output is redirected then no pbars are shown, `disable` is ignored.
2. If output is redirected but env var is specified then pbars should behave as usual.
3. If output is not redirected then env var is ignored and pbars behave as usual.

So:
```
if <env-var> or not <output-redirected>:
    ... behave as `disable` says
else: 
    ... ignore `disable`, do not show anything
```

You are now consulting disable before isatty, which is confusing.

Reply:
changed to use `TQDM_ISATTY` env var as requested.